<a href="https://colab.research.google.com/github/marciapaulasv-coder/TechChalenge/blob/main/03_modelagem.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

X_train = pd.read_csv('X_train.csv')
X_test = pd.read_csv('X_test.csv')

y_train = pd.read_csv('y_train.csv').squeeze()
y_test = pd.read_csv('y_test.csv').squeeze()

In [2]:
print('X_train:', X_train.shape)
print('X_test:', X_test.shape)
print('y_train:', y_train.shape)
print('y_test:', y_test.shape)

X_train: (25519, 47)
X_test: (10938, 47)
y_train: (25519,)
y_test: (10938,)


**1) Regressão Logistica**

In [3]:
from sklearn.linear_model import LogisticRegression
from sklearn.impute import SimpleImputer
import numpy as np

# Create an imputer to handle missing values
imputer = SimpleImputer(strategy='mean')

# Fit the imputer on X_train and transform both X_train and X_test
X_train_imputed = imputer.fit_transform(X_train)
X_test_imputed = imputer.transform(X_test)

# Convert the imputed arrays back to DataFrames, retaining column names
X_train = pd.DataFrame(X_train_imputed, columns=X_train.columns)
X_test = pd.DataFrame(X_test_imputed, columns=X_test.columns)

# Criação e treinamento do modelo de Regressão Logística
modelo_lr = LogisticRegression(random_state=42, max_iter=1000)

modelo_lr.fit(X_train, y_train)

LogisticRegression(max_iter=1000, random_state=42)

In [4]:
modelo_lr.fit(X_train, y_train)

LogisticRegression(max_iter=1000, random_state=42)

In [5]:
# Previsão na base de teste

y_pred_lr = modelo_lr.predict(X_test)

In [6]:
# Avaliação inicial da Regressão Logística

from sklearn.metrics import classification_report

print(classification_report(y_test, y_pred_lr))

              precision    recall  f1-score   support

           0       0.88      1.00      0.94      9651
           1       0.00      0.00      0.00      1287

    accuracy                           0.88     10938
   macro avg       0.44      0.50      0.47     10938
weighted avg       0.78      0.88      0.83     10938



/usr/local/lib/python3.13/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.13/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.13/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))


**Conclusão:** Apesar da acurácia de aproximadamente 88%, o modelo não conseguiu identificar nenhum cliente da classe 1 (mau pagador), apresentando recall e F1-score iguais a zero para essa classe. O resultado evidencia o impacto do desbalanceamento do TARGET e demonstra que a acurácia, isoladamente, não é adequada para avaliar o desempenho do modelo neste problema.